### PR TITLE
snippets for the biojs registry

### PR DESCRIPTION
**Pull request by wilzbach**
Hey

this adds jHeatmap and its examples to the BioJS registry (as snippets).
## Testing

You can test this locally by using 

```
npm install
npm run sniper
```

(inside of the jheatmap folder).

And open [http://localhost:9090/snippets](http://localhost:9090/snippets)

The BioJS registry runs browserify on all packages.
Then it displays the examples (aka snippets) with a server-side version of the biojs-sniper.
Settings for the snippets are set in the `package.json`.
## Publishing

It is super-easy - just run `npm publish`. The registry will pick up your package and list + display it ;-)

Here is [an example](http://registry.biojs.net/client/#/detail/biojs-vis-msa) of how the output in the registry will look.
## Remarks

I tried to make as few modifications as possible to your source, but if you continue to work on this library it would be excellent if you could split up your code into packages/modules. Tinier modules make it way easier to reuse your codebase. A good example is generally IO stuff
Plus there are existing libraries for data sorting, filtering and aggregation - no need to reinvent the wheel ;-)

### CODE REVIEW

**Comment by jordeu** [comment]
Thank you very much Sab! 

We have merged your pull request and published it to npm registry.

**Comment by wilzbach** [comment]
Thanks :)

Hmm, that is very strange. Currently the browserify server fails because of a non-existent  "site" directory. Did you run the ant task (included in the command `npm install`)?

http://wzrd.in/bundle/jheatmap@1.1.0
http://registry.biojs.net/client/#/detail/jheatmap

**Comment by mpschr** [comment]
Hi @greenify 

I can install jheatmap via the `npm install jheatmap`, enter the node `node_modules/jheatmap` folder and have to build with `npm install` it before it works. But before the build, it both the /build and /site directories are missing. But thereafter, the http://localhost:9090/snippets/enrichment examples works flawless

When trying to load the "Enrichment" exmaple (http://workmen.biojs.net/demo/jheatmap/enrichment), there are some errors which I do not understand:

`GET http://wzrd.in/bundle/jheatmap@1.1.0 500 (Internal Server Error) enrichment:9`

`Uncaught ReferenceError: require is not defined enrichment:34
Resource interpreted as Image but transferred with MIME type text/html: "http://workmen.biojs.net/github/jheatmap/jheatmap/master/images/loading.gif". enrichment:27`

Should they be in the repository without the need of building??

**Comment by wilzbach** [comment]
tl;dr: run "npm install" and republish :)

> I can install jheatmap via the npm install jheatmap, enter the node node_modules/jheatmap folder and have to build with npm install it before it works

You shouldn't need to enter the node_modules directory. If you run "npm publish" the contents of the current folder are zipped and uploaded on npm. Is it possible that there is no build folder in the folder of your jheatmap npm package?

> But before the build, it both the /build and /site directories are missing. But thereafter, the http://localhost:9090/snippets/enrichment examples works flawless

Did you run the ant task or `npm install` in the folder which you used to publish to npm?

> When trying to load the "Enrichment" exmaple

500 means `wrzrd.in` couldn't find a file (because there is no build or site folder in the published npm package). 

> Should they be in the repository without the need of building??

wrzd.in runs browserify on your main file specified in the package.json, therefore it returns an error when this browserify task fails...

**Comment by wilzbach** [comment]
BTW You can see the full error message [here](http://wzrd.in/bundle/jheatmap@1.1.0)

**Comment by mpschr** [comment]
Ok I have made a republish with build and site dir present. I was not aware that the contents from my folder are uploaded - I thought that npm pulls the project from GitHub.

The real problem was that there was no `.npmignore` file present and npm adds one that ignores the directory build. 

I think now it works
